logs_sdk/lib/opentelemetry/sdk/logs/logger_provider.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Logs
+      # The SDK implementation of OpenTelemetry::Logs::LoggerProvider.
+      class LoggerProvider < OpenTelemetry::Logs::LoggerProvider
+        attr_reader :resource
+
+        UNEXPECTED_ERROR_MESSAGE = 'unexpected error in ' \
+          'OpenTelemetry::SDK::Logs::LoggerProvider#%s'
+
+        private_constant :UNEXPECTED_ERROR_MESSAGE
+
+        # Returns a new LoggerProvider instance.
+        #
+        # @param [optional Resource] resource The resource to associate with
+        #   new LogRecords created by {Logger}s created by this LoggerProvider.
+        #
+        # @return [OpenTelemetry::SDK::Logs::LoggerProvider]
+        def initialize(resource: OpenTelemetry::SDK::Resources::Resource.create)
+          @log_record_processors = []
+          @mutex = Mutex.new
+          @resource = resource
+          @stopped = false
+        end
+
+        # Creates an {OpenTelemetry::SDK::Logs::Logger} instance.
+        #
+        # @param [String] name Instrumentation package name
+        # @param [optional String] version Instrumentation package version
```

Comment:
Minor: [YARD documentation's type parser](https://yardoc.org/types.html) does not support the "optional" marker in the type description. Perhaps the word can be moved to the descriptive string?

Reply:
```suggestion
        # @param [String] version Optional instrumentation package version
```

Reply:
Thanks for the catch, @olleolleolle! This type parser is new to me. Out of curiosity, what happens if the type parser fails? I was able to generate docs that display the type with "optional" present.

For example: <img width="778" alt="Screenshot 2023-11-07 at 11 33 44 AM" src="https://github.com/open-telemetry/opentelemetry-ruby/assets/87386821/c410afe3-b42c-451c-8160-b88b76288ada">

Reply:
Good point!

I received requests from one of opentelemetry-ruby's maintainers to revert the changes that move `optional` into the description and out of the type declaration. 

This project has a consistent pattern of using `[optional Type]` in the argument documentation. Rather than make the change in this PR, we should consider updating the standard for the entire project in one go. I've created https://github.com/open-telemetry/opentelemetry-ruby/issues/1553 for this discussion.